tools/jtest/jtest.cc

```diff
@@ -2118,7 +2118,7 @@ extract_urls(char *buf, int buflen, char *base_url)
         char *he = strchr(base_url + 8, '/');
         if (!memchr(base_url, '.', he - base_url)) {
           char t[512];
-          strcpy(t, base_url);
+          strncpy(t, base_url, sizeof(t) - 1);
```

Comment:
So `t` is the destination. Additionally, if `base_url` was not NULL terminated before then this changes nothing. What this fix does do is not overrun `t`.